code/pagetypes/Forum.php

```diff
@@ -732,8 +732,9 @@ function BBTags() {
 	 * @return Form Returns the post message form
 	 */
 	function PostMessageForm($addMode = false, $post = false) {
+		$addMode = (strtolower($addMode) === 'true' || $addMode === true) ? true : false;
```

Comment:
Personally, I'd be more inclined to make the default "true". At the moment, if someone passes in '1' as their argument, it will be interpreted as false.

As PHP is loosely typed, I think I'd probably go with something like this:

```php
if (!$addMode || strtolower($addMode) == 'false') {
    $addModel = false;
}
```

Strict type checking isn't needed here and has the added benefit that a string of value '0' will evaluate as false too :)